WalletWasabi/Wallets/SmartWallet.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin.Protocol;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore;
+using WalletWasabi.Blockchain.Analysis.FeesEstimation;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Blockchain.TransactionProcessing;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.Helpers;
+using WalletWasabi.Models;
+using WalletWasabi.Services;
+using WalletWasabi.Stores;
+
+namespace WalletWasabi.Wallets
+{
+	public class SmartWallet : BackgroundService
```

Comment:
Here we are forcing `SmartWallet` to be a `BackgroundService` when it is clear that it is not. `BackgroundService` represents a long-running task defined in the `ExecuteAsync`. This `SmartWallet` is s long-living instance but not a long-running task.

Reply:
It's long running (or it will be.) It's tricky, because the coinjoin client is a long running task, it's just we use task.run. The coinjoin client is in wallet service and the wallet service is in smart wallet. So technically it is long running, so we better start encapsulating it with the background service.